app/services/romm.py

```
@@ -0,0 +1,401 @@
+
+        headers = {"Accept": "application/json", "User-Agent": _USER_AGENT}
+        if content_type:
+            headers["Content-Type"] = content_type
+        token = self._explicit_token if self._explicit_token is not None else _token()
+        if authenticated and token:
+            # RomM client API tokens ("rmm_" + 64 hex) are sent as bearer
+            # credentials. RomM's CSRF middleware short-circuits on a bearer or
+            # basic scheme, so no CSRF token is needed for writes.
+            headers["Authorization"] = f"Bearer {token}"
+
+        req = urllib.request.Request(url, headers=headers, data=data, method=method)
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)